datastore/adapter/logging.py

```diff
@@ -16,69 +16,85 @@
 
 
 
-class _Adapter(typing.Generic[DS]):
+class _Adapter(typing.Generic[DS, RT, RV]):
 	"""Wraps a datastore with a logging shim."""
+	__slots__ = ()
 	
 	logger: logging.Logger
 	
 	
-	@typing.no_type_check
 	def __init__(self, *args, logger: logging.Logger = ROOT_LOGGER, **kwargs):
 		self.logger = logger
-		super().__init__(*args, **kwargs)
+		super().__init__(*args, **kwargs)  # type: ignore[call-arg] # noqa: F821
 	
 	
-	@typing.no_type_check
 	async def get(self, key: datastore.Key) -> RT:
 		"""Return the object named by key or None if it does not exist.
 		   LoggingDatastore logs the access.
 		"""
 		self.logger.info('%s: get %s' % (self, key))
-		value = await super().get(key)
+		value = await super().get(key)  # type: ignore[misc] # noqa: F821
+		self.logger.debug('%s: %s' % (self, value))
+		return value
+	
+	
+	async def get_all(self, key: datastore.Key) -> RV:
+		"""Return the object named by key or None if it does not exist.
+		   LoggingDatastore logs the access.
+		"""
+		self.logger.info('%s: get %s' % (self, key))
```

Comment:
Why no f-string?

Reply:
Cause that code is from 1900 and f-strings weren't invented back then :wink: 

Will fix!

Reply:
Changed it to the parameter classic string style as that will cause the string to only be composed if it will actually be logged.

Reply:
👍